### DISCUSSION TITLE
Test crawler and report issue.

### DISCUSSION THREAD

**Discussion by kcubeterm** [kcubeterm/achoz#9] [General]
### procedure.

Add single or multiple directory in DirToIndex array in config.json. 
and then call crawler.js with node. 
`node crawler.js` 

It will output a single file name Indexofdata.jsonl once crawler scanned all files and directories which has specified in config.json. 

IndexofData.jsonl contains 1 json object per line. 
Json objects contains following data.
```json
{
    "filename": "sample.pdf",
    "abspath": "/root/sample.pdf",
    "atime":      "atime ofthat file",
    "mtime":     "mtime..",
    "ctime":       "ctime"
    "content":    "extracted text from files",
    "filetype":     "types or mimetype"
}
```

**Comment by cauerego** [comment]
to be edited

https://github.com/kcubeterm/achoz/releases/tag/2021firstalpha

downloading it now...

**Comment by cauerego** [comment]
not good... 🤣

gotta continue it another time.

![Screenshot_20210728-171337](https://user-images.githubusercontent.com/560418/127358537-34d5c174-747e-425b-9264-c4e93c99f4f9.png)

```
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module 'officeparser'
Require stack:
- /storage/emulated/0/achoz/run/crawler/crawler.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/storage/emulated/0/achoz/run/crawler/crawler.js:4:22)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/storage/emulated/0/achoz/run/crawler/crawler.js' ]
}
```

**Reply by kcubeterm** [reply]
Lol. Who will perform `npm install `

**Reply by cauerego** [reply]
i tried `node install` 🤣

not used with node or shells still, i tell you!

let's see...

**Reply by cauerego** [reply]
very long `pkg update` as well...

and then even longer to npm...
```
20:14:09  1861  run  npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
[       ...........] / postinstall: sill install executeActions
```

removing the lock file also took a long time, so then i waited... until this error:

![Screenshot_20210728-204234](https://user-images.githubusercontent.com/560418/127385681-a08210fa-fe02-426a-8c48-c8a64e8b961b.png)

**Reply by cauerego** [reply]
any idea what's `he`?

[2021-07-28T19_45_33_759Z-debug.log](https://github.com/kcubeterm/achoz/files/6896094/2021-07-28T19_45_33_759Z-debug.log)

i should probably stop asking stupid questions... html encoder.

**Reply by cauerego** [reply]
well, this is a nasty issue.

it's obviously not about permission, i guess.

i'll try it more tomorrow.

**Reply by kcubeterm** [reply]
May i know why are you keeping source code in interal storage where symlink and unix file permissions has restrictions. Drag it into termux home and do npm install

**Reply by kcubeterm** [reply]
> it's obviously not about permission, i guess.

It's permission issue. Symbolic link is not allowed in internel storage of android.

**Reply by cauerego** [reply]
lol, i hate that... and always forget it.

i try to avoid using termux storage as it's not syncthinged and it's easier to lose...

came here to report i was trying this now 
https://blog.dniccumdesign.com/troubleshooting-nodejs-compatibility-issues/

> To fix this, simply remove any prepended ^ or ~ marks within your package.json file, purge the node_modules directory and the retrying the install and run commands.

still a good hint to follow...

**Reply by cauerego** [reply]
error looks much better already!

![Screenshot_20210729-224846](https://user-images.githubusercontent.com/560418/127570312-7f37801b-b7aa-47dd-80cf-3041f3a602ae.png)

**Comment by cauerego** [comment]
not a big fan of https://semver.org/

but i might read it later to see if they have a point or not.

big releases should happen once per year and have distinct names. the rest is for dev only and would probably be better served with a simple full iso timestamp.

**Reply by kcubeterm** [reply]
I follow similar semantic
